services/MerchStoreService.ts

```diff
@@ -154,19 +177,24 @@ export default class MerchStoreService {
       const item = await merchItemRepository.findByUuid(uuid);
       if (!item) throw new NotFoundError();
       const hasBeenOrdered = await Repositories.merchOrderItem(txn).hasItemBeenOrdered(uuid);
-      if (hasBeenOrdered) throw new UserError('This item has been ordered already');
+      if (hasBeenOrdered) throw new UserError('This item has been ordered and cannot be deleted');
       return merchItemRepository.deleteMerchItem(item);
     });
   }
 
   public async createItemOption(item: Uuid, option: MerchItemOption): Promise<PublicMerchItemOption> {
     return this.transactions.readWrite(async (txn) => {
       const merchItem = await Repositories.merchStoreItem(txn).findByUuid(item);
-      if (!merchItem) throw new NotFoundError('Item not found');
+      if (!merchItem) throw new NotFoundError('Merch item not found');
+      if (!merchItem.hasVariantsEnabled) throw new UserError('Cannot add option to items with variants disabled');
+      const hasDifferentOptionType = merchItem.options[0].metadata?.type !== option.metadata?.type;
```

Comment:
`metadata` should always be present right? So do we need the question marks?

Reply:
Not necessarily - I think we're missing an edge case for if we have a merch item with no variants is created but then its option is deleted with `deleteItemOption()`. When `createItemOption` is called to add a new item option (with no metadata, since `hasVariantsEnabled = false`), then line 189 prevents it from being added. I don't think the zero-option behavior is defined currently (i.e. should we let that happen? or not?), so I can ask Steven about that, or let me know what you think as well

Reply:
Then we should throw in `deleteItemOption` if we're trying to delete the only option for an item, since that leaves the item in a bad state. Items should always have at least one option IMO.

Reply:
I think one case where it should be fine to have 0 options is when it's hidden - I'm thinking within the admin UI, an admin should be able to freely edit options without having to be constrained to always keeping 1 option even if the item's hidden